src/recon_test/test_DataSymmetriesForBins_PET_CartesianGrid.cxx

```diff
@@ -124,14 +124,24 @@ run_tests_2_proj_matrices_1_bin(const ProjMatrixByBin& proj_matrix_no_symm,
 #endif
       elems_no_sym.sort();
       elems_with_sym.sort();
-      if (!check(elems_no_sym == elems_with_sym, "comparing lors"))
+      if (!check(elems_no_sym == elems_with_sym, "comparing lors") ||
+          !check(elems_with_sym.get_bin() == elems_no_sym.get_bin(), "Comparing Bin"))
 	{
 	  // SYM const Bin bin=*bin_iter;
-	  cerr << "Current bin:  segment = " << bin.segment_num() 
-		     << ", axial pos " << bin.axial_pos_num()
-		     << ", view = " << bin.view_num() 
-	       << ", tangential_pos_num = " << bin.tangential_pos_num() << "\n";
-	    ProjMatrixElemsForOneBin::const_iterator no_sym_iter= elems_no_sym.begin(); 
+
+      cerr << "Current bin: \tsegment = " << bin.segment_num()
+           << ", \taxial pos " << bin.axial_pos_num()
+           << ", \tview = " << bin.view_num()
+           << ", \ttangential_pos_num = " << bin.tangential_pos_num()
+           << "\nSymm bin: \t\tsegment = " << elems_with_sym.get_bin().segment_num()
+           << ", \taxial pos " << elems_with_sym.get_bin().axial_pos_num()
+           << ", \tview = " << elems_with_sym.get_bin().view_num()
+           << ", \ttangential_pos_num = " << elems_with_sym.get_bin().tangential_pos_num() << "\n\n";
+
+        if (elems_no_sym != elems_with_sym)
+        {
+        std::cerr << "No Symmetries Iterator || Symmetries Iterator " << std::endl;
+	    ProjMatrixElemsForOneBin::const_iterator no_sym_iter= elems_no_sym.begin();
 	    ProjMatrixElemsForOneBin::const_iterator with_sym_iter = elems_with_sym.begin();
 	    while (no_sym_iter!= elems_no_sym.end() || with_sym_iter!=elems_with_sym.end())
 	      {
```

Comment:
This check, and while loop, can give a error due to the first element of the `elems_with_sym` not being in the `elems_no_sym` list, but is close (this is noticed when I am trying a scanner with 0 tilt). An example error is when `{2, 5, -8}` exists in `elems_with_sym`(right) but is never matched in the `elems_no_sym`? ` {2, 4, -8}`, second right entry, does exist in `elems_no_sym`, as do the vast majority of tested cases.

```
Error. comparing lors
Current bin: 	segment = -1, 	axial pos 0, 	view = 112, 	tangential_pos_num = 0
Symm bin: 		segment = -1, 	axial pos 0, 	view = 112, 	tangential_pos_num = 0

No Symmetries Iterator || Symmetries Iterator 
{2, -4, 6}:0.50568    ||   
{2, -4, 7}:1.15499    ||   
{2, -4, 8}:0.339834    ||   
{2, -3, 4}:0.1962    ||   
{2, -3, 5}:1.15499    ||   
{2, -3, 6}:0.649314    ||   
{2, -2, 3}:1.04171    ||   
{2, -2, 4}:0.958793    ||   
{2, -1, 1}:0.732236    ||   
{2, -1, 2}:1.15499    ||   
{2, -1, 3}:0.113278    ||   
{2, 0, -1}:0.422757    ||   
{2, 0, 0}:1.15499    ||   
{2, 0, 1}:0.422757    ||   
{2, 1, -3}:0.113278    ||   
{2, 1, -2}:1.15499    ||   
{2, 1, -1}:0.732236    ||   
{2, 2, -4}:0.958792    ||   
{2, 2, -3}:1.04172    ||   
{2, 3, -6}:0.649313    ||   
{2, 3, -5}:1.15499    ||   
{2, 3, -4}:0.196201    ||   
{2, 4, -8}:0.339833    ||   
{2, 4, -7}:1.15499    ||   
{2, 4, -6}:0.50568    ||   
                       ||   {2, 5, -8}:0.815161
                       ||   {2, 4, -8}:0.339832
                       ||   {2, 4, -7}:1.15499
                       ||   {2, 4, -6}:0.505682
                       ||   {2, 3, -6}:0.649312
                       ||   {2, 3, -5}:1.15499
                       ||   {2, 3, -4}:0.196202
                       ||   {2, 2, -4}:0.958791
                       ||   {2, 2, -3}:1.04172
                       ||   {2, 1, -3}:0.113277
                       ||   {2, 1, -2}:1.15499
                       ||   {2, 1, -1}:0.732237
                       ||   {2, 0, -1}:0.422756
                       ||   {2, 0, 0}:1.15499
                       ||   {2, 0, 1}:0.422758
                       ||   {2, -1, 1}:0.732235
                       ||   {2, -1, 2}:1.15499
                       ||   {2, -1, 3}:0.113279
                       ||   {2, -2, 3}:1.04171
                       ||   {2, -2, 4}:0.958794
                       ||   {2, -3, 4}:0.196199
                       ||   {2, -3, 5}:1.15499
                       ||   {2, -3, 6}:0.649315
                       ||   {2, -4, 6}:0.505678
                       ||   {2, -4, 7}:1.15499
                       ||   {2, -4, 8}:0.339836
                       ||   {2, -5, 8}:0.815157
```

Reply:
ok. pretty weird that this happens now and not before but fine. There is a difficulty with symmetries for the first/last voxel, which I thought we had ok, but maybe not. Could explain some of the "circular FOV" problems.

In any case, this test would first need to sort the rows (luckily, we do have a `sort` method). Can you paste some after sorting? Ideally it's only the end bits.

If that's the case, I'd be interested in seeing what happens if you give it an odd-sized images.

(apologies, this is turning into a project on its own)

Reply:
I'm confused as we do call sort() higher up. certainly doesn't look like it

Reply:
> pretty weird that this happens now and not before but fine. 

Currently view mashing > 1 (so tilt). Set view mashing to 1 involves a lot more views/angles (error occurs at view 112) instead of the 8 currently tested.

The `sort` method is used above. It is working correctly but `{2, 5, -8}` exists on the right and not the left. Some more debugging is showing that there are 11 unmatched elements for symmetry but only 9 for no symmetry. All no symmetry coords exist in the symmetries, which happens to have an additional two.

Reply:
To clairify, unmatched in the current test although there is significant overlap between the sets. Just a few outliers.

Reply:
I'm sorry. I seem to be very stupid today. The left column (no sym) increases in 'y' (2nd coord), the right column (sym) decreases in 'y'. how can that be with the `sort`?

> there are 11 unmatched elements for symmetry but only 9 for no symmetry. All no symmetry coords exist in the symmetries, which happens to have an additional two.

completely lost. sorry. I guess you mean in 11 unmatches in the right column (use the matrix with symmetries) and 9 in the left (matrix without symmetries). but then you say 9 of them exist, so they match anyway?

Reply:
Sorry. Let me try to explain again. There is a `sort` that arranges them somewhat such that the coordinate order makes some sense. As it loops through the vectors, it tests if the iterates are equal. However, in the example above there is an additional entry `{2, 5, -8}`. Given the current `while` loop, this then stalls the iterate on the RHS from progressing until LHS has reached `end()`. 

Could use is something like [`std::set_difference`](https://www.cplusplus.com/reference/algorithm/set_difference) and look for discrepancies? However, this requires C++ 17/20 and two uses (A-B) and (B-A).

Reply:
still, the sort didn't quite sort. Obviously, this doesn't have anything to do with `2,5,-8` missing, but it seems to point to yet another problem. either with how I think `sort` works
https://github.com/UCL/STIR/blob/5daf0d6d2684bd6d1b14b4d2f6eaede92cdc283f/src/recon_buildblock/ProjMatrixElemsForOneBin.cxx#L150
or
https://github.com/UCL/STIR/blob/877511b607a90cbc3c3faebf9fa4e2dc3ea72f1b/src/include/stir/recon_buildblock/ProjMatrixElemsForOneBinValue.inl#L127-L131

Reply:
I'd like to understand the `sort` problem more, because in some places we do rely on it (e.g. in `merge` I believe). Possibly I'm just missing something obvious.

Reply:
I see what you mean now. Okay lets take sometime to talk about this tomorrow.

Reply:
Resolving and moving this to an issue https://github.com/UCL/STIR/issues/1002